03 Events/01 Events/README.md

```diff
@@ -31,7 +33,147 @@ Summary steps:
 
 ```
 
--
+- For this sample is not necessary to work with Node.js `v8.x` so we could:
+
+```bash
+nvm use 6.11.5
+```
+
+- In Node.js, _all objects that emit [`events`](https://nodejs.org/docs/latest-v6.x/api/events.html) are instances of the `EventEmitter` class. These objects expose an `eventEmitter.on(`) function that allows one or more functions to be attached to named events emitted by the object._
+
+- So, we have to require the `EventEmitter` from `events` to create an instance of:
+
+### ./index.js
+
+```diff
++ const EventEmitter = require('events').EventEmitter;
+
+```
+
+- Create an instace of `EventEmitter`:
+
+### ./index.js
+
+```diff
+const EventEmitter = require('events').EventEmitter;
+
++ const dataRetriever = (countTries) => {
++   const emitter = new EventEmitter();
++   deferredProcess(countTries, emitter);
+
++   return emitter;
++ };
+
+```
+
+- Now, we are going to simulate an async process to retrieve data from some server. We could use `process.nextTick` and `setInterval` to simulate that behaviour:
+
+### ./index.js
+
+```diff
+const EventEmitter = require('events').EventEmitter;
+
++ const deferredProcess = (countTries, emitter) => {
++   process.nextTick(() => {
++     let count = 0;
++     emitter.emit('start');
++     const interval = setInterval(() => {
++       emitter.emit('data', ++count);
++       if (count === countTries) {
++         emitter.emit('end', count);
++         clearInterval(interval);
++       }
++     }, 300);
++   });
++ };
+
+const dataRetriever = (countTries) => {
+  const emitter = new EventEmitter();
+  deferredProcess(countTries, emitter);
+
+  return emitter;
+};
+
+```
+
+- For example, we are going to retrieve `10 countries`:
```

Comment:
They are tries not countTries

Reply:
Yes, I change it on next sample. I'm going to update this one too

Reply:
OK

Reply:
Done